File: Data Structures & Algorithms/remove-element/submission-0.py
class Solution:
    def removeElement(self, a: List[int], val: int) -> int:
        l,n = 0,len(a)-1
        while(l<=n):
            if a[l]==val:
                if a[n]!=val:
                    a[l]=a[n]
                    n-=1
                    l+=1
                else:
                    n-=1
            else:
                l+=1
        return l

File: Data Structures & Algorithms/lowest-common-ancestor-of-a-binary-tree/submission-0.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
TreeNode* fun(TreeNode*root,int& p,int& q){
    if(root == nullptr || root->val == p || root->val == q)return root;
    TreeNode* left=fun(root->left,p,q);
    TreeNode* right=fun(root->right,p,q);
    if(left && right)return root;
    if(left)return left;
    return right;
}
    TreeNode* lowestCommonAncestor(TreeNode* root, TreeNode* p, TreeNode* q) {
        return fun(root,p->val,q->val);
    }
};